### ISSUE TITLE
Don't use an interactive rebase when rebasing onto another branch

### ISSUE THREAD

**Issue by jesseduffield** [jesseduffield/lazygit#2442]
**Is your feature request related to a problem? Please describe.**
Currently we always interactive rebase when rebasing onto another branch. This is not obvious behaviour and causes surprising behaviour (as we saw in https://github.com/jesseduffield/lazygit/pull/2429).

**Describe the solution you'd like**
We should not pass --interactive when rebasing onto a branch, BUT we should _still_ show the commits that are yet to come in the rebase, because the visual aid is handy.

**Comment by phillipwood** [comment]
As far as I can remember `rebase.autosqaush` is the only config setting that changes the behavior of `git rebase --interactive` relative to a plain `git rebase`. Since v2.26.0 the merge backend used by `--interactive` has been the default [1] so depending on which versions of git lazygit wants to support you could drop the `--interactive` and still be able to read the todo list.

[1] git/git@2ac0d6273fa (rebase: change the default backend from "am" to "merge", 2020-02-15)

**Comment by phillipwood** [comment]
> Just to confirm: it's ok to treat a non-interactive rebase the same as an interactive one once it stopped at a conflict, right? Including allowing the user to reorder or drop todos, or turn a pick into a fixup? Basically allowing everything they could do using `git rebase --edit-todo` at that point.

Yes, provided the rebase was not started with one of the options like `-C` that imply `--apply`. Thinking about it again it might be better to replace `--interactive` with `--merge`  as the latter has used the interactive backend since v2.21.0 [1] and it will error out if the user passes `-C`. While I cannot guarantee it I do not anticipate any changes that would stop `git rebase --edit-todo` working with a rebase started with `--merge`. 

[1] git/git@68aa495b59 (rebase: implement --merge via the interactive machinery, 2018-12-11)

**Comment by stefanhaller** [comment]
> Thinking about it again it might be better to replace `--interactive` with `--merge`

Thanks, makes sense. However, I wonder whether it's worth making that change. You said earlier that `rebase.autosquash` is the only difference, and we took care of that in #2429, so at this point it doesn't seem to matter whether we continue to use `--interactive --no-autosquash` or switch to `--merge`.
